ui/app/hooks/tests/useRetryTransaction.test.js

```diff
@@ -0,0 +1,125 @@
+import * as reactRedux from 'react-redux'
+import assert from 'assert'
+import { renderHook } from '@testing-library/react-hooks'
+import sinon from 'sinon'
+import transactions from '../../../../test/data/transaction-data.json'
+import * as methodDataHook from '../useMethodData'
+import * as metricEventHook from '../useMetricEvent'
+import { showSidebar } from '../../store/actions'
+import { useRetryTransaction } from '../useRetryTransaction'
+
+let useDispatch, useMethodData, useMetricEvent
+
+const dispatch = sinon.spy(() => Promise.resolve({ blockTime: 0 }))
+const trackEvent = sinon.spy()
```

Comment:
Can we move these into the describe block scope instead of module scope and maybe also use a [sandbox](https://sinonjs.org/releases/v9.0.2/sandbox/) to simplify the restoration?

Reply:
I definitely agree re: describe-scope, but I don't follow why using a sandbox would be simpler. Calling `sinon.restore()` in `afterEach` (thus resetting the "global sandbox") should be our default.

I've been following that pattern in an attempt to work toward being able to use a global `afterEach` hook for our entire test suite, [as recommended by the sinon docs](https://sinonjs.org/releases/v9.0.2/general-setup/), to completely eliminate the possibility of us forgetting to restore.

Reply:
Oh, that works too